.github/workflows/build_and_test.yaml

```diff
@@ -23,47 +44,67 @@ jobs:
+      - name: Restore cache
+        uses: actions/cache/restore@v3
+        id: restore-cache
         with:
-          path: cache
+          path: cache  # keep in sync with "Store cache"
           key: compile-${{ env.branch_name_for_cache }}-${{ github.sha }}
           restore-keys: |
             compile-${{ env.branch_name_for_cache }}-
             compile-
-      - name: Checkout everest-core
-        uses: actions/checkout@v3
+      - name: Checkout from `everest-core` the `.ci/` directory (reusable workflow mode, see https://stackoverflow.com/a/74123003/1168315)
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            .ci
+          repository: EVerest/everest-core
+          ref: CI/reusable-workflow  # TODO get reference in general, see https://stackoverflow.com/questions/74784735 resp. https://github.com/actions/toolkit/issues/1264
+      - name: Checkout repo ${{ inputs.repository || 'EVerest/everest-core' }} to directory `source/`
+        uses: actions/checkout@v4
         with:
-          path: source
+          path: source/
+          repository: ${{ inputs.repository }}
+          ref: ${{ inputs.repository_ref }}
       - name: Run clang-format
         uses: everest/everest-ci/github-actions/run-clang-format@v1.0.0
         with:
           source-dir: source
           extensions: hpp,cpp
           exclude: cache
-      - name: Setup run scripts
+      - name: Setup run scripts  # FIXME update `build-kit` to not use the `scripts/`-dir implicitly
         run: |
           mkdir scripts
-          rsync -a source/.ci/build-kit/ scripts
+          rsync -a .ci/build-kit/ scripts
       - name: Pull docker container
         run: |
           docker pull --quiet ghcr.io/everest/build-kit-alpine:latest
           docker image tag ghcr.io/everest/build-kit-alpine:latest build-kit
       - name: Compile
+        env:
+          CMAKE_FLAGS_EXTRA: ${{ inputs.cmake_flags_extra || '-DEVC_ENABLE_CCACHE=1 -DISO15118_2_GENERATE_AND_INSTALL_CERTIFICATES=OFF' }}
         run: |
           docker run \
           --volume "$(pwd):/ext" \
           --name compile-container \
           build-kit run-script compile
-      - name: Unit tests and install
+      - name: Unit tests
+        if: ${{ inputs.unit_tests || false }}
         run: |
-          docker commit compile-container build-image
+          docker commit compile-container unit-tests-image
           docker run \
           --volume "$(pwd):/ext" \
           --name test-container \
-          build-image run-script test_and_install
-      - name: Run integration tests
+          unit-tests-image run-script unit_tests
+      - name: Integration tests
+        if: ${{ inputs.integration_tests || true }}
         run: |
-          docker commit test-container integration-image
-          pushd source/.ci/e2e
-          docker-compose run e2e-test-server run-script tests
+          docker commit compile-container integration-tests-image
```

Comment:
Do the integration-tests rely on the unit-tests somehow?

Reply:
I don't think so :)